recipes/libx264/all/conanfile.py

```diff
@@ -60,11 +62,15 @@ def _configure_autotools(self):
                 args.append('--enable-shared')
             else:
                 args.append('--enable-static')
-            if self.settings.os != 'Windows' and self.options.fPIC:
+            if self.settings.os != 'Windows' and self.options.get_safe("fPIC"):
```

Comment:
```suggestion
            if self.options.get_safe("fPIC"):
```

Reply:
it's only needed for non-windows

Reply:
but fPIC is removed if Windows, it's useless to test Windows again

Reply:
Default is `None` as I recall, which on shared it's always deleted and the `if` would be false

```suggestion
            if self.settings.os != 'Windows' and self.options.get_safe("fPIC", True):
```

This is a super confusing setting to manage... please correct me if I am mistaken

Reply:
> but fPIC is removed if Windows, it's useless to test Windows again

it might be removed, but it's removed in another part of code, which is not so obvious to notice, so I prefer explicit for now.

Reply:
> This is a super confusing setting to manage... please correct me if I am mistaken

no idea, to be honest. in my mind, if some option doesn't exist, it means the option doesn't make any sense.
like for instance, `shared` doesn't apply if `header_only=True` for many recipes, like there are no libraries at all...
that you're talking means deleted options may actually mean `option=True`... which sounds incorrect, and I would prefer if you just discard `shared=True fPIC=False` by the `validate` or another means.

Reply:
FYI to clarify, my intention was not to change this condition. it already looks too fragile and hard to cook correctly...
that I did - is just to fix an annoying hook that didn't exist previously, so I had to add the code block
```
def configure(self):
    if self.options.shared:
        del self.options.fPIC
```
which is no mandatory for some reason...
and now, the condition we had before breaks:
```
if self.settings.os != 'Windows' and self.options.fPIC:
```
because `fPIC` doesn't exist for `self.options.shared` any more.
I see two ways to fix the condition (without changing it too much):
```
if self.settings.os != 'Windows' and self.options.get_safe("fPIC"):
```
and
```
if self.settings.os != 'Windows' and not self.options.shared and self.options.fPIC:
```
and from the two above, I prefer the first approach, as it's not so bulky, and resembles what we had before more closely.
I hope it clarifies everything.
again, the intention and purpose of the PR is just to fix M1. this `get_safe` is just to pass the hook.

Reply:
The motivation for deleting `fPIC` is because it makes no sense when `shared` ... it must be `True` for the shared object to be usable...

However the older recipes like this one assumed false which now should be true

As must as I love your suggestion, 

```py
if self.settings.os != 'Windows' and not self.options.shared and self.options.fPIC:
```

Will never set `--enable-fpic` when `shared` so the library would be useful...
My personal rule, with our hooks is always use `self.options.get_safe("fPIC", True)`

Reply:
yes, maybe it needs to have default True or default False to get proper settings. again, that's not a point of PR to improve fPIC management.
for fPIC, I am already super confused by contradictory suggestions...
I actually need help with another hook (see my first comment) and would be glad if you can suggest a fix for that weird error :)
we didn't have that error before, but I am sure I didn't change anything related to Windows build, and I am sure the library should be there (otherwise, it wouldn't link).